kratos/processes/check_same_modelpart_using_skin_distance_process.h

```diff
@@ -0,0 +1,225 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "containers/model.h"
+#include "includes/kratos_parameters.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class CheckSameModelPartUsingSkinDistanceProcess
+ * @ingroup KratosCore
+ * @brief Checks that the model part is the same using the skin distance
+ * @details Using skin distance, this process checks that the model part is the same
+ * @author Vicente Mataix Ferrandiz
+ */
+template<std::size_t TDim>
+class KRATOS_API(KRATOS_CORE) CheckSameModelPartUsingSkinDistanceProcess
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of CheckSameModelPartUsingSkinDistanceProcess
+    KRATOS_CLASS_POINTER_DEFINITION(CheckSameModelPartUsingSkinDistanceProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    explicit CheckSameModelPartUsingSkinDistanceProcess(
+        Model& rModel,
+        Parameters ThisParameters = Parameters(R"({})")
+        )
+        : mrModel(rModel),
+          mThisParameters(ThisParameters)
+    {
+        KRATOS_TRY
+
+        Parameters default_parameters = GetDefaultParameters();
+        mThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+        KRATOS_CATCH("");
+    }
+
+    /// Destructor.
+    ~CheckSameModelPartUsingSkinDistanceProcess() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void operator()()
+    {
+        Execute();
+    }
```

Comment:
```suggestion
```
I don't think this is a good idea. I'd stick to the `Execute` only to make clear what is being done.